### PR TITLE
ccw4601 remove name from reshape op

### PR DESCRIPTION
**Pull request by nke001**
This pull request solves all the original issues in  https://github.com/Theano/Theano/pull/4777#discussion_r72066004. This one is correctly rebased, so the history should not be messy.

close gh-4601

### CODE REVIEW

**Comment by lamblin** [comment]
Jenkins test this please

**Comment by nke001** [comment]
@nouiz @adri-romsor

File: theano/gpuarray/opt.py [hunk 1]
```diff
@@ -643,10 +643,7 @@ def local_gpua_contiguous(op, context_name, inputs, outputs):
 @op_lifter([tensor.Reshape])
 @register_opt2([tensor.Reshape], 'fast_compile')
 def local_gpua_reshape(op, context_name, inputs, outputs):
-    name = op.name
-    if name:
-        name = 'Gpu' + name
-    res = GpuReshape(op.ndim, op.name)
+    res = GpuReshape(op.ndim, None)
```
Comment (abergeron) [inline theano/gpuarray/opt.py]: You should probably not pass None either since you removed the argument from Reshape().

**Comment by abergeron** [comment]
jenkins test this please

File: theano/tensor/basic.py [hunk 2]
```diff
@@ -4607,7 +4607,7 @@ def reshape(x, newshape, ndim=None, name=None):
                 "to know what the number of dimensions of the reshaped "
                 "variable will be. You can provide the 'ndim' keyword "
                 "argument to 'reshape' to avoid this problem." % newshape)
-    op = Reshape(ndim, name)
```
Comment (nouiz) [inline theano/tensor/basic.py]: You need to modify the class Reshape to assert that the name it receive is None and don't set it as an attribute.

**Comment by nke001** [comment]
@nouiz Hi, could you start Jerkins to test the latest commit? Thanks

**Comment by lamblin** [comment]
Jenkins test this please

**Comment by nke001** [comment]
@nouiz @adri-romsor All tests have been passed, it seems like we can merge the pull request, thanks.

**Pull request merged**